core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -406,6 +406,26 @@ class TransactionsTest extends KafkaServerTestHarness {
     TestUtils.waitUntilTrue(() => offsetAndMetadata.equals(consumer.committed(Set(tp).asJava).get(tp)), "cannot read committed offset")
   }
 
+  @Test(expected = classOf[TimeoutException])
+  def testSendOffsetsToTransactionTimeout(): Unit = {
+    val producer = createTransactionalProducer("transactionProducer", maxBlockMs = 1000)
+    producer.initTransactions()
+    producer.beginTransaction()
+    producer.send(new ProducerRecord[Array[Byte], Array[Byte]](topic1, "foo".getBytes, "bar".getBytes))
+
+    for (i <- 0 until servers.size)
+      killBroker(i)
+
+    try {
+      producer.sendOffsetsToTransaction(Map(
```

Comment:
Do we have unit test coverage for other transaction API max blocking as well? Do you mind adding them as separate tests and share the same module?

Reply:
I added some timeout tests for initTransaction, commitTransction, abortTransaction using same base method.
Is this implementation correct what you intended?